cysync/src/pages/initialApp/deviceFlow/deviceSetup/stepperFormComponents/upgradingDevice.tsx

```diff
@@ -89,11 +92,19 @@ const UpgradingDevice: React.FC<StepComponentProps> = ({
     errorObj,
     latestVersion,
     setBlockNewConnection,
-    setIsDeviceUpdating
+    setIsDeviceUpdating,
+    errorResolutionState
   } = useDeviceUpgrade(true);
 
   const refreshComponent = () => {
-    handleRetry();
+    if (
+      errorResolutionState ===
+      DeviceUpgradeErrorResolutionState.DEVICE_AUTH_REQUIRED
+    ) {
+      onClose();
+    } else {
```

Comment:
Don't do this, if Cysync installs a malicious firmware and auth fails in which case the device is compromised. Cysync will move to main menu.

Reply:
lets leave it as it is as we are not restricting the user if they have malicious firmware.